kore/src/Kore/Attribute/RuleIndex.hs

```diff
@@ -0,0 +1,63 @@
+{-|
+Module      : Kore.Attribute.RuleIndex
+Description : Rule index attribute
+Copyright   : (c) Runtime Verification, 2018
+License     : NCSA
+Maintainer  : ???@runtimeverification.com
+
+-}
+module Kore.Attribute.RuleIndex
+    ( RuleIndex (..)
+    , ruleIndexId, ruleIndexSymbol, ruleIndexAttribute
+    ) where
+
+import Control.DeepSeq
+       ( NFData )
+import Data.Default
+import GHC.Generics
+       ( Generic )
+
+import           Kore.AST.Kore
+import           Kore.Attribute.Parser
+                 ( ParseAttributes (..) )
+import qualified Kore.Attribute.Parser as Parser
+
+{- | @RuleIndex@ represents the @RuleIndex@ attribute.
+ -}
+newtype RuleIndex = RuleIndex { getRuleIndex :: Maybe Int }
+    deriving (Eq, Ord, Show, Generic)
+
+instance NFData RuleIndex
+
+instance Default RuleIndex where
+    def = RuleIndex Nothing
+
+-- | Kore identifier representing the @ruleIndex@ attribute symbol.
+ruleIndexId :: Id Object
+ruleIndexId = "ruleIndex"
+
+-- | Kore symbol representing the @ruleIndex@ attribute.
+ruleIndexSymbol :: SymbolOrAlias Object
+ruleIndexSymbol =
+    SymbolOrAlias
+        { symbolOrAliasConstructor = ruleIndexId
+        , symbolOrAliasParams = []
+        }
+
+-- | Kore pattern representing the @ruleIndex@ attribute.
+ruleIndexAttribute :: CommonKorePattern
+ruleIndexAttribute =
+    (asCommonKorePattern . ApplicationPattern)
+        Application
+            { applicationSymbolOrAlias = ruleIndexSymbol
+            , applicationChildren = []
+            }
+
+instance ParseAttributes RuleIndex where
```

Comment:
Would this still work without this instance and without adding it to the big `ParseAttributes` list? Since we're not really parsing it, we only care about it having its default `Nothing` value. Try removing this and the extra parser line in `Axiom.hs`